dapp/src/components/TransactionListener.js

```diff
@@ -152,10 +156,16 @@ const TransactionListener = ({ rpcProvider }) => {
     }
   }
 
-  const updateTransactions = (transactions) => {
-    const txHashes = transactions.map((t) => t.hash)
-    const otherTxs = transactions.filter((tx) => !txHashes.includes(tx.hash))
-    const newTransactions = [...otherTxs, ...transactions]
+  const updateTransactions = (transactionsToUpdate) => {
+    const txHashes = transactionsToUpdate.map((t) => t.hash)
+    /* With functional approach values inside a function scope can be out of date
```

Comment:
@tomlinton now I remember the reason I did the Tx listener in an Object oriented approach instead of functional. There were cases where multiple async functions would be executing simultaneously and sometimes values in those functions would be outdated. Lets say: 

2 routines simultaneously do some CRUD to transactions and store end result. The one finishing last will most probably override the results of the one finishing previously. Since the value of `transactions` has been copied to the function scope before the previous routine has modified it. You can see how on master (without this commit) any newer transaction overrides previous ones. 

Anyway having this component in functional approach with the use of `store.currentState` instead of OOO is still nicer IMHO.